src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -1449,13 +1449,14 @@ private decimal ToDecimal()
 
             // II) Perform multiplication
 
-            uint[] rglData1 = new uint[4] { x._data1, x._data2, x._data3, x._data4 };
-            uint[] rglData2 = new uint[4] { y._data1, y._data2, y._data3, y._data4 };
+            ReadOnlySpan<uint> rglData1 = stackalloc uint[4] { x._data1, x._data2, x._data3, x._data4 };
+            ReadOnlySpan<uint> rglData2 = stackalloc uint[4] { y._data1, y._data2, y._data3, y._data4 };
 
             //Local buffer to hold the result of multiplication.
             //Longer than CReNumeBuf because full precision of multiplication is carried out
             const int x_culNumeMultRes = 9;       // Maximum # UI4s in result buffer in multiplication
-            uint[] rgulRes = new uint[x_culNumeMultRes]; //new [] are already initialized to zero
+            Span<uint> rgulRes = stackalloc uint[x_culNumeMultRes]; //new [] are already initialized to zero
```

Comment:
I added the second comment to complement the original. You're right that it's undefined per-spec but that's unusual because all other allocated memory is zeroed or initialized to stated values. I think it's worth leaving in given that it's a comment and has no runtime cost.

Reply:
Sure, it's true there's no runtime costs, but the comments simply describe how spans work, and is fairly evident to anyone doing high-perf work. For someone new to the business, yeah, it may look weird - but I don't think that's a good reason to have a comment. But it's just a comment, if you feel strongly about it I don't mind that much.

Reply:
I'm not sure about strongly, but the reason I put it there was that I made the mistake of not clearing it and that broke a test.

You're right that anyone doing high performance work should be aware of these things but I think it can ease the cognitive load on even those people sometimes if you just explain a little about what's going on. I do a lot of code archaeology in my day job and a small comment explaining the why of something can make a big difference.